ddtrace/contrib/trace_utils.py

```diff
@@ -230,16 +231,39 @@ def ext_service(pin, int_config, default=None):
     return default
 
 
+def _identity(x):
+    return x
+
+
+def _add_if_needed(gateway, target, original_value, address, formatter=_identity):
+    if original_value is None:
+        return
+    key = address.value
+    if not gateway.is_needed(key):
+        return
+    target[key] = formatter(original_value)
+
+
 def set_http_meta(
     span,
     integration_config,
     method=None,
     url=None,
+    raw_uri=None,
     status_code=None,
     status_msg=None,
     query=None,
+    query_object=None,
+    format_query_object=_identity,
```

Comment:
Is there a reason why this formatting can't happen before or after the call to `set_http_meta`? My concern is that this new signature is making this function look like it's doing a lot of complex work. It would be great if the formatting part could be separated from the rest IMHO.

Reply:
That's a very good point. The idea I had here is to make sure we run the formater only if we need the value (for instance, looping on all headers could be expensive so we want to avoid it if not needed). Passing the formater function was a way to still keep the logic close to the instrumentation. Wdyt?

Reply:
Looking at how this is implemented, it looks like all the new processing is happening at the end of the `set_http_meta` call, and no interim computations are reused. So perhaps the new code could be the body of a new AppSec helper that can be called after `set_http_meta` where necessary?

Reply:
That works for me too. I am mostly afraid it would lead to calling two methods that look the same in a raw. I don't have strong opinions on wether this is something we want to avoid. (cc wdyt @brettlangdon @Kyle-Verhoog @mabdinur )